src/biosHandler.cpp

```diff
@@ -0,0 +1,49 @@
+#include "biosHandler.hpp"
+
+#include "logger.hpp"
+
+namespace vpd
+{
+void BiosHandler::checkAndListenPLDMService()
+{
+    // Setup a call back match on NameOwnerChanged to determine when PLDM is up.
+    static std::shared_ptr<sdbusplus::bus::match_t> nameOwnerMatch =
+        std::make_shared<sdbusplus::bus::match_t>(
+            *m_asioConn,
+            sdbusplus::bus::match::rules::nameOwnerChanged(
+                "xyz.openbmc_project.PLDM"),
+            [this](sdbusplus::message_t& msg) {
+        if (msg.is_method_error())
+        {
+            logging::logMessage("Error in reading PLDM name owner signal.");
+            return;
+        }
+
+        std::string name;
+        std::string newOwner;
+        std::string oldOwner;
+
+        msg.read(name, oldOwner, newOwner);
+        if (!newOwner.empty() && name == "xyz.openbmc_project.PLDM")
+        {
+            // restore BIOS attribute from here.
+            //  We don't need the match anymore
+            nameOwnerMatch.reset();
```

Comment:
If the vpd-manager service comes first, later PLDM service comes up. nameOwnerChanged signal will catch PLDM service coming up, this time should we have callback for handling "restoring BIOS attribute" ?

Reply:
Yes we should. Whats your opinion on that?

Reply:
yes, we should have 'restore' functionality, A service can acquires bus name at any time.